x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/execution_log_table/execution_log_columns.tsx

```diff
@@ -117,7 +117,7 @@ export const GET_EXECUTION_LOG_METRICS_COLUMNS = (
       />
     ),
     render: (value: number) => (
-      <>{value ? <RuleDurationFormat duration={value} isMillis={true} /> : getEmptyValue()}</>
+      <>{value ? <RuleDurationFormat duration={value} isSeconds={true} /> : getEmptyValue()}</>
```

Comment:
Not a big deal, but it would be better to check the value using `value != null`. Otherwise, `-` is displayed instead of `0` in the gap duration column.

![Screenshot 2022-05-02 at 18 20 54](https://user-images.githubusercontent.com/1938181/166270349-3f1bc3d3-9813-429b-8e90-712d37681f2f.png)

Reply:
Hmmm, I thought gaps were only written if `> 0` 🤔 

https://github.com/elastic/kibana/blob/3730dd0779ed8efd74aee88f57422781ec1ac122/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts#L197-L210


Ahhh, I see... This happens for gaps 1ms->499ms as we normalize `gap_duration` to a second and round which can result in `0`:

https://github.com/elastic/kibana/blob/3730dd0779ed8efd74aee88f57422781ec1ac122/x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/client_for_executors/client.ts#L154-L156

I do think it's a little weird to show `0` when the log message is showing `6ms` (no weirder than `-`), but this is the value written if they're looking in Discover, and the column tooltip does say the unit is in minutes, so this rounding could be understood that way. Nice catch! I'll make the change 🙂. (FWIW sorting should still be fine here as this is just on the rendering side.)

Reply:
Rsolved in [22cc0c8](https://github.com/elastic/kibana/pull/130072/commits/22cc0c8dbd2a1300675caf4c6d471d211ed44858)!

Reply:
So with this change successful rule executions are now showing all `0`'s for gap instead of `--` as `0` is our missing value coming from the aggs query ([source](https://github.com/elastic/kibana/blob/36234191468045dfeb5b9049a6996467538b3f91/x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/event_log/get_execution_event_aggregation/index.ts#L183-L187)). In discussion with @xcrzx, we decided to revert to the old behavior as it seems better to show `--` for `1ms-499ms` gap durations instead of `0` for successful ones that don't actually have a gap (better represented by `--`).


<img width="1465" alt="Rules_-_Kibana" src="https://user-images.githubusercontent.com/2946766/166501897-f174665e-f103-4137-a3d3-fad85defbcb0.png">


Additionally, I noticed I named the field `gap_duration_ms` when it's actually in `seconds` so I will rename all these to `gap_duration_s` as well 👍